### PR TITLE
add default for Arm Compiler

### PR DESCRIPTION
**Pull request by juntangc**
Purpose: Add compiler option for armclang/armflang.

TYPE: enhancement

KEYWORDS: Arm Compiler for Linux

SOURCE: Jun Tang (Amazon)

DESCRIPTION OF CHANGES:
 Add compile defaults for Arm Compiler for Linux.

LIST OF MODIFIED FILES: 
M       arch/configure.defaults

TESTS CONDUCTED: 
1. Tested the latest Arm compiler with this configuration. It shows 3% performance improvement on CONUS 12km benchmark. 
2. The Jenkins tests are all passing.

RELEASE NOTE: This PR added compiler option for armclang/armflang.

### CODE REVIEW

**Comment by weiwangncar** [comment]
The regression test results:

    Test Type              | Expected  | Received |  Failed
    = = = = = = = = = = = = = = = = = = = = = = = =  = = = =
    Number of Tests        : 23           24
    Number of Builds       : 60           57
    Number of Simulations  : 158           150        0
    Number of Comparisons  : 95           86        0

    Failed Simulations are: 
    None
    Which comparisons are not bit-for-bit: 
    None

**Comment by weiwangncar** [comment]
@mgduda  @islas Should we consider this for 4.5.1?

**Comment by islas** [comment]
@weiwangncar That seems good to me. @mgduda , any objections?

File: arch/configure.defaults [hunk 1]
```diff
@@ -2123,6 +2123,46 @@ NETCDFPAR_BUILD =       CONFIGURE_NETCDFPAR_BUILD
...
+SCC             =      armclang
+CCOMP           =      armclang
+DM_FC           =      mpif90
+DM_CC           =      mpicc -DMPI2_SUPPORT
+FC              =      CONFIGURE_FC
+CC              =      CONFIGURE_CC
+LD              =      $(FC)
+RWORDSIZE       =      CONFIGURE_RWORDSIZE
+PROMOTION       =
+ARCH_LOCAL      =
```
Comment (islas) [inline arch/configure.defaults:2144]: @mgduda Would we need `-DAARCH64_X86_CORRECTNESS_FIX` in the `ARCH_LOCAL` defines?

Comment (juntangc) [reply arch/configure.defaults:2144]: The flag "-DAARCH64_X86_CORRECTNESS_FIX" was added for GCC to provide a way to achieve same results using GCC on x86 and aarch64 platforms. We don't need this flag here.

**Comment by juntangc** [comment]
@weiwangncar please take a look at this PR.  thanks,

File: arch/configure.defaults [hunk 2]
```diff
@@ -2123,6 +2123,46 @@ NETCDFPAR_BUILD =       CONFIGURE_NETCDFPAR_BUILD
 
 #insert new stanza here
 
+###########################################################
+#ARCH    Linux aarch64, GCC compiler OpenMPI # serial smpar dmpar dm+sm
```
[the remark this answers is not in the document]
Comment (juntangc) [reply arch/configure.defaults]: Done.

**Review by islas — APPROVED** [review]

**Review by weiwangncar — APPROVED** [review]

**Comment by weiwangncar** [comment]
@islas Will need approval again after resolving conflict...

**Review by islas — APPROVED** [review]

**Review by weiwangncar — APPROVED** [review]

**Pull request merged**